components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml

```diff
@@ -4693,6 +4693,177 @@ document.observe("xwiki:dom:loaded", function() {
+    </property>
+    <property>
+      <code>var PhenoTips = (function(PhenoTips) {
+  var widgets = PhenoTips.widgets = PhenoTips.widgets || {};
+  widgets.SegmentedBar = Class.create({
+    DEFAULT_SEGMENTS : 5,
+    DEFAULT_FILL : '#777',
+    DEFAULT_BORDER : '#777',
+    DEFAULT_SEGMENT_HEIGHT : '5px',
+    DEFAULT_DISPLAY_PERCENT_VALUE_FLAG : true,
+    /**
+    A PhenoTips widget for creating a segment bar html element. Example:
+
+        var barOptions = {
+          segments : 10,
+          fill: '#625',
+          border : '#987',
+          segmentHeight : '10px',
+          segmentWidth : '3px',
+          displayPercentValueFlag : false
+        };
+        container.insert(new PhenoTips.widgets.SegmentedBar(score, barOptions).generateSegmentedBar());
+
+    @param value The value that this bar is meant to display. Must be between 0 and 1.
+
+    @param options Options for styling the segmented bar, all values are optional:
+
+    {
+      segments: The number of segments in the bar. Default is 5.,
+      fill: The fill color for the segments. Default is #777.,
+      border: The border color of the segments. Default is #000.,
+      segmentHeight: The height of each individual segment. Default is 5px.,
+      segmentWidth: The width of each individual segment. Default is equal to either the specified height or the default height.,
+      displayPercentValueFlag: Option to display the percentage value after the bar. Default is true
+    }
+
+    */
+    initialize : function(value, options) {
+      if (value &gt; 1 || value &lt; 0) {
+        console.log("Invalid segmented bar value");
+      }
+      this.value = value;
+      this.options = options || {};
+      this.segments = this.options.segments || this.DEFAULT_SEGMENTS;
+      this.fill = this.options.fill || this.DEFAULT_FILL;
+      this.border = this.options.border || this.DEFAULT_BORDER;
+      this.segmentHeight = this.options.segmentHeight || this.DEFAULT_SEGMENT_HEIGHT;
+      this.segmentWidth = (this.options.segmentWidth || this.options.segmentHeight) || this.DEFAULT_SEGMENT_HEIGHT;
+      this.displayPercentValueFlag = (this.options.displayPercentValue != undefined) ? this.options.displayPercentValue : this.DEFAULT_DISPLAY_PERCENT_VALUE_FLAG;
+    },
+    generateSegmentedBar: function() {
+      var bar = new Element('div', {
+        'class' : 'segmented-bar',
+        'style' : 'color: ' + this.border + ';',
+        'title': Math.round(this.value * 100) + '%' || ''
+      });
+      var valueUnit = 1 / this.segments;
+      for (var i = 0; i &lt; this.segments; ++i) {
+        var segmentFill = Math.round(100 * Math.min(Math.max((this.value - i * valueUnit) / valueUnit, 0), 1));
+        var segment = new Element('span', {
+          'class': 'segmented-unit',
+          'style': 'height: ' + this.segmentHeight +';' +
+                    'width: ' + this.segmentWidth + ';' +
+                    'border-color:' + this.border + ';'
+        }).update(
```

Comment:
It is possible to do this without the need for an inner span.

If we don't separate between the border and fill colors, it's enough to set border-left-width to 1+box width * segmentFill.

If we do want different border and fill colors, then this will work:

```
.setStyle({
  backgroundClip: 'content-box', // This goes in the CSS below
  boxSizing: 'border-box', // This goes in the CSS below
  padding: 0, // This goes in the CSS below
  backgroundColor: this.fill,
  borderColor: this.border,
  paddingRight: (1-segmentFill) * this.segmentWidth,
  height: this.segmentHeight + 1 + 'px', // Assuming the width is just a value without a unit, otherwise use:
  width: 'calc(' + this.segmentWidth + '+1px)'
});